syncmaster/backend/celery.py

```diff
@@ -0,0 +1,6 @@
+# SPDX-FileCopyrightText: 2023-2024 MTS PJSC
+# SPDX-License-Identifier: Apache-2.0
+from syncmaster.backend.settings import ServerAppSettings
+from syncmaster.worker import celery_factory
+
+app = celery_factory(ServerAppSettings())
```

Comment:
We don't use `python -m celery -A syncmaster.backend.celery` here, so why this is global?

Reply:
Could we add separated Celery factories for server & scheduler, so they won't import anything from `syncmaster.worker`?

Reply:
> We don't use `python -m celery -A syncmaster.backend.celery` here, so why this is a global variable created during module initialization?

We could create Celery instance in application factory and pass it via `Depends()`, but we also have to mock Celery in unit tests somehow

Reply:
You could use something like that:
```
@fixture
def celery() -> Celery: ...


@fixture
def test_app(celery) -> FastAPI:
  app = application_factory()
  app.dependency_overrides[Celery] = lambda: app
  return app
```

Reply:
> Could we add separated Celery factories for server & scheduler, so they won't import anything from `syncmaster.worker`?

1. We will still have to use `imports=["syncmaster.worker.transfer"]` in each instance
2. Implementation of `task_cls` (like `WorkerTask`) will be needed as our current tasks rely on `engine` and `settings.encryption.crypto_key`

Reply:
1. Are those imported anyway or just when celery worker receives a task to execute?
2. This is required only on worker side, not server/scheduler

Reply:
Found the reason why it didn't work, fixed